api/catalog/api/serializers/base.py

```diff
@@ -0,0 +1,19 @@
+import re
+
+from django.conf import settings
+from rest_framework import serializers
+
+
+class SchemableHyperlinkedIdentityField(serializers.HyperlinkedIdentityField):
+    """
+    This field returns the link but allows the option to replace the URL scheme.
+    """
+
+    def __init__(self, scheme=settings.API_LINK_SCHEME, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.scheme = scheme
+
+    def get_url(self, *args, **kwargs):
+        url = super().get_url(*args, **kwargs)
+        return re.sub(r"\w+://", f"{self.scheme}://", url, 1)
```

Comment:
I'd recommend anchoring this regex search, it'll be faster and will prevent anything that's later in the URL from being accidentally updated (as unlikely as that is), although I guess the `count=1` would also prevent that.
```suggestion
        return re.sub(r"^\w+://", f"{self.scheme}://", url, 1)
```

Reply:
Cool idea, committed in 5839c3e.